### PR TITLE
feat: per-task git worktree isolation

### PR DESCRIPTION
**Pull request by ApprenticeGC**
## Summary
- Each task gets an isolated git worktree (`.worktrees/swarm-{taskId}`) so concurrent builders don't corrupt each other's workspace state
- Worktree path flows end-to-end: `ExecuteRoleTask.WorkspacePath` → `SubscriptionCliRoleExecutor` (sets `WorkingDirectory` on CLI processes + `{{workspace}}` template var) → `SandboxCommandBuilder`
- `BuildVerifier` (Verify GOAP step) and `GitArtifactCollector` also run scoped to the task's worktree
- `WorkspaceBranchManager` extended with `EnsureWorktreeAsync()` / `RemoveWorktreeAsync()` lifecycle
- New config: `WorktreeIsolationEnabled` (requires `WorkspaceBranchEnabled: true`)
- Enabled in `appsettings.Local.json` for dogfooding
- 5 new tests (583 total, 0 failures)

## Motivation
During RFC-006 dogfooding, concurrent builder tasks all shared the same workspace. The Verify step (`dotnet build` + `dotnet test`) ran against the main workspace where all tasks' changes were interleaved, making build verification unreliable. Worktree isolation ensures each task operates on a clean, independent copy of the repository.

## Files changed (11)
| File | Change |
|------|--------|
| `InternalMessages.cs` | Added `WorkspacePath` to `ExecuteRoleTask` |
| `TaskCoordinatorActor.cs` | Worktree creation in Build, path passed to Review/Rework/Verify, cleanup in PostStop |
| `RuntimeOptions.cs` | Added `WorktreeIsolationEnabled` |
| `WorkspaceBranchManager.cs` | `EnsureWorktreeAsync`, `RemoveWorktreeAsync`, `RunGitAsync` helper |
| `SubscriptionCliRoleExecutor.cs` | `effectiveWorkspace`, `WorkingDirectory` on processes, `{{workspace}}` var |
| `BuildVerifier.cs` | `workingDirectory` parameter for build+test commands |
| `GitArtifactCollector.cs` | `workspacePath` parameter for git commands |
| `Worker.cs` | Pass `WorktreeIsolationEnabled` to `WorkspaceBranchManager` |
| `appsettings.Local.json` | Enable `WorkspaceBranchEnabled` + `WorktreeIsolationEnabled` |
| 2 test files | 5 new tests for worktree isolation behavior |

## Test plan
- [x] Build succeeds (0 errors)
- [x] All 583 tests pass (578 existing + 5 new)
- [x] Pre-commit hooks pass
- [ ] Dogfood with concurrent task submission to verify worktree isolation works end-to-end

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Introduced per-task workspace isolation using git worktrees to prevent concurrent Git state conflicts during task execution.
  * Added a configuration option to enable workspace isolation across all tasks.

* **Tests**
  * Added test coverage for workspace isolation functionality and edge cases.

### CODE REVIEW

File: project/dotnet/src/SwarmAssistant.Runtime/Execution/SubscriptionCliRoleExecutor.cs [hunk 1]
```diff
@@ -137,6 +137,8 @@ private async Task<CliRoleExecutionResult> ExecuteCoreAsync(ExecuteRoleTask comm
         var errors = new List<string>();
         // Use pre-built prompt if provided (with code context), otherwise build from scratch
         var prompt = command.Prompt ?? RolePromptFactory.BuildPrompt(command);
+        // Per-task worktree path takes priority over global workspace
+        var effectiveWorkspace = command.WorkspacePath ?? _workspacePath;
```
[the remark this answers is not in the document]
Comment (ApprenticeGC) [reply project/dotnet/src/SwarmAssistant.Runtime/Execution/SubscriptionCliRoleExecutor.cs]: Fixed in 2763f05. Switched from ?? to string.IsNullOrWhiteSpace check.

File: project/dotnet/src/SwarmAssistant.Runtime/Execution/WorkspaceBranchManager.cs [hunk 2]
```diff
@@ -25,6 +27,84 @@ public static string BranchNameForTask(string taskId)
...
+        var worktreePath = Path.Combine(repoRoot, ".worktrees", branchName.Replace('/', '-'));
+
+        try
+        {
+            await RunGitAsync(["worktree", "remove", worktreePath, "--force"]);
+            _logger?.LogInformation("Worktree removed path={WorktreePath} taskId={TaskId}", worktreePath, taskId);
+
+            // Clean up the branch too
+            await RunGitAsync(["branch", "-D", branchName]);
+            _logger?.LogDebug("Branch deleted branch={Branch} taskId={TaskId}", branchName, taskId);
```
[the remark this answers is not in the document]
Comment (ApprenticeGC) [reply project/dotnet/src/SwarmAssistant.Runtime/Execution/WorkspaceBranchManager.cs]: Fixed in 2763f05. RemoveWorktreeAsync now checks exit codes — logs LogWarning on worktree remove failure, LogDebug on branch delete failure.
